backend/src/main/java/gov/cdc/usds/simplereport/service/DeviceTypeService.java

```diff
@@ -189,6 +226,228 @@ public DeviceType createDeviceType(CreateDeviceType createDevice) {
     return dt;
   }
 
+  public String extractSpecimenTypeCode(String specimenDescription) {
+    Pattern specimenCode = Pattern.compile("^(.*?)\\^");
+    return extractSpecimenDetails(specimenDescription, specimenCode);
+  }
+
+  public String extractSpecimenTypeName(String specimenDescription) {
+    Pattern specimenName = Pattern.compile("\\^(.*?)\\^");
+    return extractSpecimenDetails(specimenDescription, specimenName);
+  }
+
+  private String extractSpecimenDetails(String specimenDescription, Pattern specimenName) {
+    Pattern specimenDetails = Pattern.compile("\\(([^)]*)\\)[^(]*$");
+
+    Matcher matcher = specimenDetails.matcher(specimenDescription);
+
+    if (!matcher.find()) {
+      return "";
+    }
+
+    var result = specimenName.matcher(matcher.group(1));
+
+    if (!result.find()) {
+      return "";
+    }
+
+    return result.group(1);
+  }
+
+  public List<UUID> getSpecimenTypeIdsFromDescription(LIVDResponse device) {
+    return device.getVendorSpecimenDescription().stream()
+        .map(this::parseVendorSpecimenDescription)
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .map(SpecimenType::getInternalId)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Wrapper method for syncing devices from LIVD table so automation can call the inner method
+   * without hitting the lock or conditions.
+   */
+  @Scheduled(cron = "0 * * * * *", zone = "America/New_York")
+  @SchedulerLock(
+      name = "DeviceTypeService_syncDevices",
+      lockAtLeastFor = "PT30S",
+      lockAtMostFor = "PT30M")
+  @ConditionalOnProperty("simple-report.device-sync-livd.enabled")
+  public void scheduledSyncDevices() {
+    syncDevices();
+  }
+
+  @Transactional
+  @AuthorizationConfiguration.RequireGlobalAdminUser
+  public void syncDevices() {
+    List<LIVDResponse> devices = client.getLIVDTable();
+
+    var devicesToSync = new HashMap<DeviceType, ArrayList<SupportedDiseaseTestPerformedInput>>();
+    var deviceSpecimens = new HashMap<String, List<UUID>>();
+
+    devices.forEach(
+        device -> {
+          // Skip sync for any device entries with incomplete data
+          if (device == null
+              || device.getManufacturer() == null
+              || device.getModel() == null
+              || device.getVendorAnalyteName() == null) {
+            log.info("One or more required values are not present in the LIVD entry");
```

Comment:
The log statements in this class were added for debugging purposes - some are definitely useful but I'll tidy these up a bit before merging.